documentation.md

```diff
@@ -0,0 +1,521 @@
+  <figcaption>Fig: In-depth burst storage, for a burst length of 4</figcaption>
+</figure>
+
+<figure class="image">
+  <img src="https://i.imgur.com/fzaWZ6v.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: In-depth burst storage addressing</figcaption>
+</figure>
+
+Therefore, there are _MaxBurstLen_ times as many elementary RAM cells in _i_payload_ram_ than there are in each of _i_meta_ram_out_tail_ and _i_meta_ram_out_head_.
+
+An extended cell is called _active_ if it has already acquired a response and has not acquired and released all the responses corresponding to the allocated burst.
+
+### Linked list implementation
+
+Linked lists enforce the ordering of the response release as defined by the AXI identifier.
+
+#### Linked list pointers
+
+Linked lists are logical structures maintained by the _i_meta_ram_out_tail_ and _i_meta_ram_out_head_ RAMs as well as four pointers:
+
+- Reservation head (_rsv_heads_q_): Points to the most recently reserved extended cell.
+- Response head (_rsp_heads_): Points to the next RAM address where a response of the corresponding AXI identifier will be stored.
+- Pre*tail (\_pre_tails*): Points to the second-to-last cell hosting or awaiting a response in the linked list.
+- Tail (_tails_): Points to the oldest cell hosting or awaiting a response in the linked list.
+
+The order of the pointers must always be respected. They can be equal but never overtake each other, in the order defined by the linked list.
+
+Two distinct tail pointers are required to dynamically manage the two following cases:
+
+- The pre_tail address is given as input to the payload RAM if there is a successful output handshake and the value at the output has an AXI id corresponding to the AXI id of the considered linked list. It must point to:
+  - The second-to-last element of the linked list if the list contains two reponses or more,
+  - The last element of the linked list if the list contains just one response,
+  - rsp_head if the list contains no responses.
+- The tail address is given as input to the payload RAM in all other cases. This case disjunction prevents an output data from being output twice, and prevents any bandwidth drop at the output. It must point to:
+
+  - The last element of the linked list if the list contains one response or more,
+  - pre_tail if the list contains no responses.
+
+<figure class="image">
+  <img src="https://i.imgur.com/Y7XwICc.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Linked list representation. t: tail, pt: pre_tail, rsp: response head, rsv: reservation head. Arrows between extended cells represent the linked list structure (cells must not have consecutive addresses)</figcaption>
+</figure>
+ 
+ 
+#### Lengths
+
+In addition to the pointers, lengths of sub-segments of linked lists are stored to maintain the state of each linked lists, which is not fully defined by the four pointer and metadata RAMs only:
```

Comment:
You should explain or give an example why they are not fully defined by the four pointers because looking at the figure it seems they would be fully defined by the pointers.

Reply:
I added an example, and added that if we maintain the current burst management, we may not need the lengths anymore. I put information about it in this part and also in the Future work section. Thanks!